### PR TITLE
Add gihub action build workflow

### PR DESCRIPTION
**Pull request by rmaz**
First step for resolving #12, add a Github action workflow. Requires #11  to build.

### CODE REVIEW

**Comment by rmaz** [comment]
Triggered an example here: https://github.com/rmaz/zld/runs/481515248

Seems to work fine, could probably do with some refining later on.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+name: Build
+jobs:
+  build:
+    name: Build
+    runs-on: macos-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Xcode select
+        run: sudo xcode-select -s /Applications/Xcode_11.2.app/Contents/Developer
```
Comment (michaeleisel) [inline .github/workflows/build.yml]: so this step is telling the CI process which version of xcode it should use to build? we can assume that the CI server has this copy of xcode at this path?

**Comment by michaeleisel** [comment]
this won't block merge for now, right? we can manually inspect any failures until it proves stable

File: .github/workflows/build.yml (hunk 1 above)
Comment (rmaz) [reply .github/workflows/build.yml]: Correct, you can see the list of installed software here: https://github.com/actions/virtual-environments/blob/master/images/macos/macos-10.15-Readme.md

**Comment by rmaz** [comment]
> this won't block merge for now, right? we can manually inspect any failures until it proves stable

Right, you would have to configure that yourself as described here: https://help.github.com/en/github/administering-a-repository/enabling-required-status-checks

**Comment by michaeleisel** [comment]
Looks like we can merge unless you object

**Comment by rmaz** [comment]
Good to go from my side

**Comment by michaeleisel** [comment]
Great, thank you!

**Pull request merged**